src/lib/components/FieldSelect.svelte

```diff
@@ -64,6 +64,7 @@
 	}
 
 	function handleOpenChange(wasMenuOpened: boolean) {
+		return;
```

Comment:
@fmaclen about this, what is the purpose of this function?

I am experiencing this bug, which I was able to fix by adding an early return to the `hanleOpenChange` function. It seems that when I select an option and I open the menu again but I don't change my selection, then the `FieldSelect` value is set to undefined, but the stored value remains the same.

https://github.com/user-attachments/assets/9bed30f2-ce18-4ec0-9196-0aae4af2bd7b

Reply:
I think the logic in `handleOpenChange` was there so that you could press `Escape` with the opened menu and it would clear the field. We have a bunch of assertions for this behavior (which are currently failing):

```typescript
// tests/ui.test.ts:79

// Click outside the combobox to apply the cleared value
await page.keyboard.press('Escape');
await expect(modelCombobox).toHaveValue('');
await expect(modelCombobox).toHaveAttribute('placeholder', 'Available models');
await expect(page.getByRole('option')).toHaveCount(0);
await expect(page.getByTitle('Clear')).not.toBeVisible();
```

I think the bug you are seeing is caused by this logic:

```typescript
$: if ($knowledgeStore)
	$knowledgeOptions = $knowledgeStore?.filter(
		(k) => !$attachments.find((a) => a.knowledge?.id === k.id)
	);
```

If instead of `$knowledgeOptions` we use the `$knowledgeStore` directly in the `KnowledgeSelect` everything works as expected, including highlighting the selected Knowledge:

![image](https://github.com/user-attachments/assets/4518da14-e5de-498e-9c52-b503691ab41b)

Well, _"everything works"_ with the exception that now we don't filter out the "used" knowledge in the other fields, which definitely is a nice feature.

I think the solution would be to do something like this:

```svelte
<KnowledgeSelect
	value={attachment.knowledge?.id}
	options={$knowledgeStore?.filter((k) => 
		// Only filter out knowledge that's selected in OTHER attachments
		!$attachments.find((a) => 
			a.fieldId !== attachment.fieldId && // Skip current attachment
			a.knowledge?.id === k.id
		)
	)}
	showLabel={false}
	fieldId={`attachment-${attachment.fieldId}`}
	onChange={(knowledgeId) =>
		knowledgeId && handleSelectKnowledge(attachment.fieldId, knowledgeId)}
	allowClear={false}
/>
```

This way we can now get rid of `let knowledgeOptions: Writable<Knowledge[]> = writable([]);` completely.

Reply:
Thank you! It worked like a charm. I removed `knowledgeOptions` and the code looks cleaner now.